From a33d089c1d2b669ee932df9a8b29236934102aad Mon Sep 17 00:00:00 2001
From: Christian Persch <chpe@src.gnome.org>
Date: Sun, 13 Apr 2025 09:54:39 +0200
Subject: [PATCH] emulation: Fix OSC 111

Fixes a regression from commit 9d4e38343c93bc0866997dfe4ebdc12d4e1ae7a8

Thanks to Egmont Koblinger for the git bisect and finding the bug.

Fixes: https://gitlab.gnome.org/GNOME/vte/-/issues/2876
(cherry picked from commit 39950ac5a0f32e431c53f9c8f43c818fd8c988a4)
---
 src/vteseq.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/vteseq.cc b/src/vteseq.cc
index 174298e3d..5bdec69f7 100644
--- a/src/vteseq.cc
+++ b/src/vteseq.cc
@@ -7663,7 +7663,7 @@ Terminal::OSC(vte::parser::Sequence const& seq)
                 break;
 
         case VTE_OSC_XTERM_RESET_COLOR_TEXT_BG:
-                reset_color(ColorPaletteIndex::default_fg(), ColorSource::Escape);
+                reset_color(ColorPaletteIndex::default_bg(), ColorSource::Escape);
                 break;
 
         case VTE_OSC_XTERM_RESET_COLOR_CURSOR_BG:
-- 
GitLab

